# A file cannot be named "index" (no-index)

Prevents files from being named "index".

## Rule Details

This rule aims to prevent files from being named "index", which will lead to files having meaningful names.

Examples of **incorrect** filename for this rule:

```sh
index.js
index.ts
```

Examples of **correct** filename for this rule:

```sh
calculatePrice.js
login.tsx
```
